edb/edgeql/compiler/conflicts.py

```diff
@@ -593,8 +593,8 @@ def compile_insert_unless_conflict_on(
 
 def _has_explicit_id_write(stmt: irast.MutatingStmt) -> bool:
     for elem, _ in stmt.subject.shape:
-        assert elem.rptr is not None
-        if elem.rptr.ptrref.shortname.name == 'id':
+        assert isinstance(elem.expr, irast.Pointer)
```

Comment:
This is an aside, and not a change request for this PR, but we should be able to make `shape` contain `SetE[Pointer]`

Reply:
Oh wait, that's already done and we can drop this assert